os/mm/mm_heap/mm_mallinfo.c

```diff
@@ -73,9 +87,112 @@
  * Private Functions
  ****************************************************************************/
 
+static void dump_node(struct mm_allocnode_s *node, node_type_t type)
+{
+	if (type == TYPE_CORRUPTED) {
+		mdbg("CORRUPTED NODE: addr = 0x%08x size = %u preceding size = %u\n", node, node->size, MM_PREV_NODE_SIZE(node));
+	} else if (type == TYPE_OVERFLOWED) {
+		mdbg("OVERFLOWED NODE: addr = 0x%08x size = %u type = %c\n", node, node->size, IS_ALLOCATED_NODE(node) ? 'A':'F');
+	}
+#ifdef CONFIG_DEBUG_MM_HEAPINFO
+	mdbg("Node owner pid = %u, allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+#endif
+}
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+
+/****************************************************************************
+ * Name: mm_check_heap_corruption
+ *
+ * Description:
+ *   This function walk through heap and prints information about corrupt node.
+ ****************************************************************************/
+int mm_check_heap_corruption(struct mm_heap_s *heap)
+{
+	struct mm_allocnode_s *node;
+	struct mm_allocnode_s *prev;
+	struct mm_allocnode_s *next;
+#if CONFIG_KMM_REGIONS > 1
+	int region;
+#else
+#define region 0
+#endif
+
+	DEBUGASSERT(heap);
+
+	/* Visit each region */
+
+#if CONFIG_KMM_REGIONS > 1
+	for (region = 0; region < heap->mm_nregions; region++)
+#endif
+	{
+		/* Visit each node in the region
+		 * Retake the semaphore for each region to reduce latencies
+		 */
+
+		mm_takesemaphore(heap);
+
+		prev = NULL;
+		node = heap->mm_heapstart[region];
+		next = (struct mm_allocnode_s *)((char *)node + node->size);
+
+		for (; node < heap->mm_heapend[region]; prev = node, node = next, next = (struct mm_allocnode_s *)((char *)next + next->size)) {
+			if (prev && prev->size != MM_PREV_NODE_SIZE(node)) {
```

Comment:
All branch return -1, you'd better use below style.
```
if (a) {
   return -1;
}

if (b) {
   return -1;
}
...
```

Reply:
Depth is not quiet deep here, so I agree with your answer.
But please understand that above style is very good habit for readability.